cmd/sink-worker/main.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"syscall"
+	"time"
+
+	health "github.com/AppsFlyer/go-sundheit"
+	healthhttp "github.com/AppsFlyer/go-sundheit/http"
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/go-slog/otelslog"
+	"github.com/oklog/run"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sagikazarmark/slog-shim"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+
+	"github.com/openmeterio/openmeter/config"
+	"github.com/openmeterio/openmeter/internal/dedupe"
+	"github.com/openmeterio/openmeter/internal/sink"
+	"github.com/openmeterio/openmeter/pkg/gosundheit"
+)
+
+func main() {
+	v, flags := viper.New(), pflag.NewFlagSet("OpenMeter", pflag.ExitOnError)
+	ctx := context.Background()
+
+	config.Configure(v, flags)
+
+	flags.String("config", "", "Configuration file")
+	flags.Bool("version", false, "Show version information")
+
+	_ = flags.Parse(os.Args[1:])
+
+	if v, _ := flags.GetBool("version"); v {
+		fmt.Printf("%s version %s (%s) built on %s\n", "Open Meter", version, revision, revisionDate)
+
+		os.Exit(0)
+	}
+
+	if c, _ := flags.GetString("config"); c != "" {
+		v.SetConfigFile(c)
+	}
+
+	err := v.ReadInConfig()
+	if err != nil && !errors.As(err, &viper.ConfigFileNotFoundError{}) {
+		panic(err)
+	}
+
+	var conf config.Configuration
+	err = v.Unmarshal(&conf, viper.DecodeHook(config.DecodeHook()))
+	if err != nil {
+		panic(err)
+	}
+
+	err = conf.Validate()
+	if err != nil {
+		panic(err)
+	}
+
+	extraResources, _ := resource.New(
+		context.Background(),
+		resource.WithContainer(),
+		resource.WithAttributes(
+			semconv.ServiceName("openmeter"),
```

Comment:
Service name should be unique, so it should be something like `openmeter-sink-worker`.